File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 3/FestivalMuzicaJava/src/main/java/repository/ShowsDBRepository.java
package repository;

import domain.Show;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class ShowsDBRepository implements ShowsRepository {

    private final String TABLE_NAME = "shows";
    private final Logger logger = LogManager.getLogger();
    private final JdbcUtils jdbcUtils;

    public ShowsDBRepository(JdbcUtils jdbcUtils) {
        this.jdbcUtils = jdbcUtils;
    }

    @Override
    public int size() {
        logger.traceEntry();
        int result = 0;

        try (Connection connection = jdbcUtils.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(
                    String.format("SELECT COUNT(*) FROM %s", TABLE_NAME)
            );
            logger.info("Executing: " + statement);

            ResultSet resultSet = statement.executeQuery();
            if (resultSet.next())
                result = resultSet.getInt(1);
        } catch (SQLException e) {
            logger.error(e);
        }

        logger.traceExit(result);
        return result;
    }

    @Override
    public void save(Show entity) {
        logger.traceEntry();

        try (Connection connection = jdbcUtils.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(
                    String.format("INSERT INTO %s (artist_name, date, location, remaining_seats) VALUES (?, ?, ?, ?)", TABLE_NAME)
            );

            statement.setString(1, entity.getArtistName());
            statement.setDate(2, java.sql.Date.valueOf(entity.getDate()));
            statement.setString(3, entity.getLocation());
            statement.setInt(4, entity.getRemainingSeats());
            statement.executeUpdate();
        } catch (SQLException e) {
            logger.error(e);
        }

        logger.traceExit();

    }

    @Override
    public void delete(Long aLong) {
        logger.traceEntry();

        try (Connection connection = jdbcUtils.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(
                    String.format("DELETE FROM %s WHERE id = ?", TABLE_NAME)
            );
            logger.info("Executing: " + statement);

            statement.setLong(1, aLong);
            statement.executeUpdate();
        } catch (SQLException e) {
            logger.error(e);
        }

        logger.traceExit();

    }

    @Override
    public void update(Long aLong, Show entity) {
        logger.traceEntry();

        try (Connection connection = jdbcUtils.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(
                    String.format("UPDATE %s SET artist_name = ?, date = ?, location = ?, remaining_seats = ? WHERE id = ?", TABLE_NAME)
            );
            logger.info("Executing: " + statement);

            statement.setString(1, entity.getArtistName());
            statement.setDate(2, java.sql.Date.valueOf(entity.getDate()));
            statement.setString(3, entity.getLocation());
            statement.setInt(4, entity.getRemainingSeats());
            statement.setLong(5, aLong);
            statement.executeUpdate();
        } catch (SQLException e) {
            logger.error(e);
        }

        logger.traceExit();

    }

    @Override
    public Show findOne(Long aLong) {
        logger.traceEntry();
        Show result = null;

        try (Connection connection = jdbcUtils.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(
                    String.format("SELECT * FROM %s WHERE id = ?", TABLE_NAME)
            );
            logger.info("Executing: " + statement);

            statement.setLong(1, aLong);
            ResultSet resultSet = statement.executeQuery();
            if (resultSet.next()) {
                result = new Show(
                        resultSet.getLong("id"),
                        resultSet.getString("artist_name"),
                        resultSet.getDate("date").toLocalDate(),
                        resultSet.getString("location"),
                        resultSet.getInt("remaining_seats")
                );
            }
        } catch (SQLException e) {
            logger.error(e);
        }

        logger.traceExit(result);
        return result;
    }

    @Override
    public Iterable<Show> findAll() {
        logger.traceEntry();
        List<Show> result = new ArrayList<>();

        try (Connection connection = jdbcUtils.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(
                    String.format("SELECT * FROM %s", TABLE_NAME)
            );
            logger.info("Executing: " + statement);

            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                Show show = new Show(
                        resultSet.getLong("id"),
                        resultSet.getString("artist_name"),
                        resultSet.getDate("date").toLocalDate(),
                        resultSet.getString("location"),
                        resultSet.getInt("remaining_seats")
                );
                result.add(show);
            }
        } catch (SQLException e) {
            logger.error(e);
        }

        logger.traceExit(result);
        return result;
    }
}

File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 5/FestivalMuzicaCSharpWFA/FestivalMuzicaCSharpWFA/Repository/ShowsDbRepository.cs
using FestivalMuzicaCSharp.Domain;
using FestivalMuzicaCSharp.Utils;
using log4net;

namespace FestivalMuzicaCSharp.Repository;

public class ShowsDbRepository : IShowsRepository
{
    private const string TableName = "shows";
    private readonly ILog Log = LogManager.GetLogger(typeof(ShowsDbRepository));
    private IDictionary<string, string> props;

    public ShowsDbRepository(IDictionary<string, string> props)
    {
        this.props = props;
    }

    public int Size()
    {
        Log.InfoFormat("Entering Size");
        var con = DBUtils.getConnection(props);
        var result = 0;

        using (var comm = con.CreateCommand())
        {
            comm.CommandText = $"select count(*) from {TableName}";

            Log.InfoFormat("Executing: {0}", comm.CommandText);

            result = Convert.ToInt32(comm.ExecuteScalar());
        }

        Log.InfoFormat("Exiting Size with value {0}", result);
        return result;
    }

    public void Save(Show entity)
    {
        Log.InfoFormat("Entering Save with value {0}", entity);
        var con = DBUtils.getConnection(props);

        using (var comm = con.CreateCommand())
        {
            comm.CommandText = $"insert into {TableName}(artist_name, date, location, remaining_seats) values(@artistName, @date, @location, @remainingSeats)";

            var paramArtistName = comm.CreateParameter();
            paramArtistName.ParameterName = "@artistName";
            paramArtistName.Value = entity.ArtistName;
            comm.Parameters.Add(paramArtistName);

            var paramDate = comm.CreateParameter();
            paramDate.ParameterName = "@date";
            paramDate.Value = entity.Date.ToDateTime(TimeOnly.MinValue);
            comm.Parameters.Add(paramDate);

            var paramLocation = comm.CreateParameter();
            paramLocation.ParameterName = "@location";
            paramLocation.Value = entity.Location;
            comm.Parameters.Add(paramLocation);

            var paramRemainingSeats = comm.CreateParameter();
            paramRemainingSeats.ParameterName = "@remainingSeats";
            paramRemainingSeats.Value = entity.RemainingSeats;
            comm.Parameters.Add(paramRemainingSeats);

            Log.InfoFormat("Executing: {0}", comm.CommandText);

            comm.ExecuteNonQuery();
        }

        Log.InfoFormat("Exiting Save");
    }

    public void Delete(long id)
    {
        Log.InfoFormat("Entering Delete with value {0}", id);
        var con = DBUtils.getConnection(props);

        using (var comm = con.CreateCommand())
        {
            comm.CommandText = $"delete from {TableName} where id = @id";

            var paramId = comm.CreateParameter();
            paramId.ParameterName = "@id";
            paramId.Value = id;
            comm.Parameters.Add(paramId);

            Log.InfoFormat("Executing: {0}", comm.CommandText);

            comm.ExecuteNonQuery();
        }

        Log.InfoFormat("Exiting Delete");
    }

    public void Update(long id, Show entity)
    {
        Log.InfoFormat("Entering Update with value {0}", entity);
        var con = DBUtils.getConnection(props);

        using (var comm = con.CreateCommand())
        {
            comm.CommandText = $"update {TableName} set artist_name = @artistName, date = @date, location = @location, remaining_seats = @remainingSeats where id = @id";

            var paramArtistName = comm.CreateParameter();
            paramArtistName.ParameterName = "@artistName";
            paramArtistName.Value = entity.ArtistName;
            comm.Parameters.Add(paramArtistName);

            var paramDate = comm.CreateParameter();
            paramDate.ParameterName = "@date";
            paramDate.Value = entity.Date.ToDateTime(TimeOnly.MinValue);
            comm.Parameters.Add(paramDate);

            var paramLocation = comm.CreateParameter();
            paramLocation.ParameterName = "@location";
            paramLocation.Value = entity.Location;
            comm.Parameters.Add(paramLocation);

            var paramRemainingSeats = comm.CreateParameter();
            paramRemainingSeats.ParameterName = "@remainingSeats";
            paramRemainingSeats.Value = entity.RemainingSeats;
            comm.Parameters.Add(paramRemainingSeats);

            var paramId = comm.CreateParameter();
            paramId.ParameterName = "@id";
            paramId.Value = id;
            comm.Parameters.Add(paramId);

            Log.InfoFormat("Executing: {0}", comm.CommandText);

            comm.ExecuteNonQuery();
        }

        Log.InfoFormat("Exiting Update");
    }

    public Show? FindOne(long id)
    {
        Log.InfoFormat("Entering FindOne with value {0}", id);
        var con = DBUtils.getConnection(props);
        Show? result = null;

        using (var comm = con.CreateCommand())
        {
            comm.CommandText = $"select * from {TableName} where id = @id";

            var paramId = comm.CreateParameter();
            paramId.ParameterName = "@id";
            paramId.Value = id;
            comm.Parameters.Add(paramId);

            Log.InfoFormat("Executing: {0}", comm.CommandText);

            using (var dataR = comm.ExecuteReader())
            {
                if (dataR.Read())
                {
                    var showId = dataR.GetInt32(0);
                    var artistName = dataR.GetString(1);
                    var date = DateOnly.FromDateTime(dataR.GetDateTime(2));
                    var location = dataR.GetString(3);
                    var remainingSeats = dataR.GetInt32(4);

                    result = new Show(showId, artistName, date, location, remainingSeats);
                }
            }
        }

        Log.InfoFormat("Exiting FindOne with value {0}", result);
        return result;
    }

    public IEnumerable<Show> FindAll()
    {
        Log.InfoFormat("Entering FindAll");
        var con = DBUtils.getConnection(props);
        var result = new List<Show>();

        using (var comm = con.CreateCommand())
        {
            comm.CommandText = $"select * from {TableName}";

            Log.InfoFormat("Executing: {0}", comm.CommandText);

            using (var dataR = comm.ExecuteReader())
            {
                while (dataR.Read())
                {
                    var showId = dataR.GetInt32(0);
                    var artistName = dataR.GetString(1);
                    var date = DateOnly.FromDateTime(dataR.GetDateTime(2));
                    var location = dataR.GetString(3);
                    var remainingSeats = dataR.GetInt32(4);

                    result.Add(new Show(showId, artistName, date, location, remainingSeats));
                }
            }
        }

        Log.InfoFormat("Exiting FindAll with value {0}", result);
        return result;
    }
}